manager/src/main/java/io/gingersnapproject/Caches.java

```diff
@@ -80,6 +100,28 @@ public Stream<String> getKeys(String name) {
             .stream();
    }
 
+   public Uni<Map<String, String>> getAll(String name, Collection<String> keys)  {
+      CacheAccessRecord<String> cacheAccessRecord = metrics.recordCacheAccess();
+      try {
+         Map<String, Uni<String>> res = getOrCreateMap(name).getAll(keys);
+
+         if (res.isEmpty()) return EMPTY_MAP_UNI;
+
+         UniJoin.Builder<KeyValuePair<String, String>> builder = Uni.join().builder();
+         for (Map.Entry<String, Uni<String>> entry : res.entrySet()) {
+            String key = entry.getKey();
+            cacheAccessRecord.localHit(entry.getValue() instanceof UniItem<String>);
```

Comment:
This won't work, we would need a separate cache access record per key.

Also this doesn't look like it completes the access. `uni.onItemOrFailure().invoke(cacheAccessRecord)`

Reply:
I delegated this to the `get` operation.